LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py

```diff
@@ -561,8 +569,16 @@ def setup_me(is_lad):
     aad_auth_url = ""
     amrurl = "https://management.azure.com/subscriptions/" + subscription_id + "?api-version=2014-04-01"
     try:
-        req = urllib2.Request(amrurl, headers={'Content-Type':'application/json'})
-        res = urllib2.urlopen(req)
+        req = urllib.request.Request(amrurl, headers={'Content-Type':'application/json'})
+
+        # urlopen alias in future backport is broken on py2.6, fails on urls with HTTPS - https://github.com/PythonCharmers/python-future/issues/167
```

Comment:
This might be a bit of a gotcha in the future; if someone else has to add another call over HTTPS and isn't aware of this, they'll reintroduce the bug. Is there any easy fix?

Reply:
This was the only 'fix' I could get working. This happens only on py2.6 due to issues with some py2.6 versions being built with older ssl libraries which makes it not support https. I think having this comment with the link to the detailed issue should be of sufficient help in diagnosis of any future bugs (if they are introduced) .